DCOS-39906: Add package version information to Service Table

juliangieseke opened the pull request:
## Testing

Verify that Version information is displayed on Services Table
<!--
What is needed to test the changes? e.g. specific cluster, service definitions
How can one see the result of your work? e.g. configurations, URLs
-->

## Trade-offs

![screen shot 2018-08-10 at 17 03 08](https://user-images.githubusercontent.com/530632/43965701-40375596-9cc0-11e8-942d-a8cb9e4ea5f9.png)

After talking to packaging team I decided to display the given version string as is, there is no way for us to parse the string and we dont want to display "wrong" information.
example: `2.0.3-4.0.0e-rc3`
`2.0.3` refers to the package version
-
`4.0.0e` is base tec version
-
`rc3` is package version again…

cc @lilysamimi @AmrAbdelrazik
---

Also I had to fix the first Commit from last PR, we do not only want to display Package vErsions for frameworks, but also the regular version string for marathon services. Luckily we're on a feature branch here, so I will rebase it after this PR got merged 👍 
<!--
Are you aware of any weak spots? e.g. performance, functionality
Did you decide anything noteworthy? e.g. algorithms, data structures, tools
-->

## Dependencies

<!--
What needs to happen before this can be merged? e.g. PRs merged, other events
-->

juliangieseke commented:
1. There was a screenshot in the feature which I took for column position 🤷‍♂️ (can we please clarify this before implementation next time? 🙇)
2. We dont have a reliable string to work on, your idea might work on the given one, but we just dont know anything about the string given from the API :( imagine something like this: **2.0.0**-_1.0.0e-alpha_-**rc2** (bold == package, italics == base tech)
3. added 👍 
4. same as 1. please 🙇 (we can change this, but in a later PR please)
5. I tried to expand the column to 300px, but because of our broken table styles, it is simply ignored 
 😠  
<img width="916" alt="screen shot 2018-08-13 at 14 39 20" src="https://user-images.githubusercontent.com/530632/44032181-b09dec1c-9f06-11e8-82a1-9974e4d5b9f6.png">

nLight commented:
@lilysamimi I'm all in for native tooltips, though maybe here we'd like to be consistent + delay until the native tooltip shows up is an eternity 😩 May I suggest using our custom tooltip that is instant and insanely great? :)

<img width="458" alt="screen shot 2018-08-13 at 17 43 16" src="https://user-images.githubusercontent.com/186223/44042382-62d50050-9f20-11e8-9f29-a1466f8dca3a.png">

nLight commented:
On the formatting note, shall we drop by SDK folks and ask them? We always can gracefully degrade to raw string. If there's an opportunity to give a nicer version, let's not let it slip away.

leemunroe commented:
@juliangieseke re. width, the widths AFAIK are defined in the `<colgroup>` — did you try adding a `<col>` for the version column?

leemunroe commented:
@juliangieseke re. (1) what screenshot were you referencing that shows the version after instances?

AmrAbdelrazik commented:
@juliangieseke @lilysamimi  and all
1- Please implement in that order
Name, Status, Version, Region, Instance, CPU, MEM, DISK, GPU
2- Please parse base tech version, do not include the SDK version in this view, for the special version of something like 2.0.3-4.0.0e-rc3, we should sync with the SDK team to make sure that there isn't many permutations. Or we can just ask them for a specific field that just captures the base tech version 
3- For marathon status, is that how it looked like before? if yes, then let's use it as it, if not then let's check with the Marathon team on a usable format

AmrAbdelrazik commented:
Chatted with Chris Lambert: 

In the Marathon environment, by convention, schedulers have the base tech version.
https://github.com/mesosphere/dcos-commons/blob/master/frameworks/hdfs/universe/marathon.json.mustache#L31

Beta packages use a `beta-` prefix on the package name, as well as a a `-beta` suffix on the version string.  Kinda overkill.
https://github.com/mesosphere/universe/tree/version-3.x/repo/packages/B/beta-cassandra/8

We will probably have to add a new field in cosmos for the package environment so that It's cleaner later

weblancaster commented on plugins/services/src/js/containers/services/ServicesTable.js:
```diff
@@ -389,6 +390,15 @@ class ServicesTable extends React.Component {
 
+  renderVersion(prop, service) {
+    if (!service.getVersion) {
+      return <span />;
```
I have seen us using `null` instead if `<span />`, not a blocker for me though (as long a empty span isn't rendered)

juliangieseke commented on plugins/services/src/js/containers/services/ServicesTable.js:
```diff
@@ -389,6 +390,15 @@ class ServicesTable extends React.Component {
   }
 
+  renderVersion(prop, service) {
+    if (!service.getVersion) {
```
I'm checking for the function, not the functions return value 👍

juliangieseke commented:
- added tooltip
- return `null` if empty
- fixed column with - missed the colgroup, thanks @leemunroe 🙇 
- @leemunroe there was a screenshot attached to the feature https://jira.mesosphere.com/browse/DCOS-39535
- @AmrAbdelrazik has pointed out a good example why we cant easily parse the string. in most cases, it might be `package-basetech`, but in some cases just stripping of the `package-` part isnt sufficient, because we have `package-basetech-packagebeta` - we also cant strip off every `-suffix` suffix, because it might be that it actually relates to the base tech 🤷‍♂️ 
Overall, parsing the version string would be based on a lot of conventions and assumptions and there is a high risk introducing errors and bugs. I'd say we either merge it as is (full version string) and figure out a good solution later (@AmrAbdelrazik can you create a new feature?) or we postpone this feature completely, if the current implementation doesnt fit the needs. 😕

Current state:
![screen recording 2018-08-14 at 12 45 pm](https://user-images.githubusercontent.com/530632/44087388-fc491632-9fbf-11e8-97d2-55907c90c872.gif)
(flickering is an issue with the recording, not in reality 🤷‍♂️ )

nLight commented:
Please include Pods as well

<img width="671" alt="screen shot 2018-08-15 at 14 56 23" src="https://user-images.githubusercontent.com/186223/44149183-66b0f216-a09b-11e8-9731-c09ceac17fbe.png">

juliangieseke commented:
Uh! Good catch, thanks 🙇 Added another fixup commit for the first PR (we merged that one too fast 😓 )
